src/main/java/me/zilzu/mycoupon/storage/CouponRepository.java

```diff
@@ -67,4 +69,22 @@ public CouponEntity delete(String id) {
         }
         return database.remove(id);
     }
+
+    public void invalidate(String couponId) {
+        CouponEntity entity = database.get(couponId);
+        entity.valid = false;
+        database.put(entity.id, entity);
+    }
+
+    public void saveCouponHistory(CouponUsageHistoryEntity entity) {
```

Comment:
```suggestion
    public void save(CouponUsageHistoryEntity entity) {
```

이렇게만 해도 충분히 의도가 표현이 되지않을까?
Argument 의 type만으로도 충분히 읽히지.

save CouponUsageHistoryEntity = CouponUsageHistoryEntity 를 저장하다.